Browser/keywords/promises.py

```diff
@@ -47,6 +48,25 @@ def promise_to(self, kw: str, *args) -> Future:
         | `Click`           \\#delayed_request
         | ${body}=        `Wait For`              ${promise}
         """
+        library_method = {
+            "wait for": self.library.wait_for,
+            "wait for alert": self.library.wait_for_alert,
+            "wait for function": self.library.wait_for_function,
+            "wait for navigation": self.library.wait_for_navigation,
+            "wait for request": self.library.wait_for_request,
+            "wait for response": self.library.wait_for_response,
+        }
+        if EXECUTION_CONTEXTS.current:
```

Comment:
This `if/else` logic does not solve the problem when library is used in the way described https://github.com/MarketSquare/robotframework-browser/issues/1685#issue-1105409364 Instead trying to solve it like this, it would better look https://github.com/robotframework/PythonLibCore/blob/main/src/robotlibcore.py direction and see how it can be solved in unified way when using library purely from Python and with Robot Framework. 

Similar problem is already solved when registering keyword to run on failure. Looking in this direction: https://github.com/MarketSquare/robotframework-browser/blob/main/Browser/browser.py#L741 and making it more generic would perhaps be the best option.